syntax = "proto3";

package example.v1;

import "buf/validate/validate.proto";

message PostalAddress {
  string line1 = 1 [(buf.validate.field).string = {
    min_len: 1
    max_len: 120
  }];
  string line2 = 2 [(buf.validate.field).string.max_len = 120];
  string city = 3 [(buf.validate.field).string = {
    min_len: 1
    max_len: 120
  }];
  string state = 4 [(buf.validate.field).string.max_len = 120];
  string postal_code = 5 [(buf.validate.field).string = {
    min_len: 3
    max_len: 20
  }];
  string country_code = 6 [(buf.validate.field).string.len = 2];
}
